components/director/internal/domain/root_resolver.go

```diff
@@ -143,6 +146,12 @@ func NewRootResolver(transact persistence.Transactioner, scopeCfgProvider *scope
 	}
 }
 
+func getHttpClient() *http.Client {
```

Comment:
Now you are modifying the default http client which may be not intended. Maybe you can create here separate instance of http client and use it for that purpose.
```go
return &http.Client{
    Timeout: 3 * time.Second,
}
```